metadata-ingestion/src/datahub/cli/ingest_cli.py

```diff
@@ -128,7 +128,15 @@ async def run_pipeline_to_completion(pipeline: Pipeline) -> int:
         logger.info("Starting metadata ingestion")
         with click_spinner.spinner(disable=no_spinner):
             try:
-                pipeline.run()
+                if pipeline.config.flags.generate_memory_profiles:
```

Comment:
just marking down a future need for refactoring
```suggestion
                # TODO: Move this logic to pipeline.py.
                if pipeline.config.flags.generate_memory_profiles:
```

Reply:
Moved the code.